### DISCUSSION TITLE
zod and fields relation

### DISCUSSION THREAD

**Discussion by vaagnavanesyan** [react-hook-form/react-hook-form#8467] [Q&A] [answered]
```
"react-hook-form": "7.29.0",
"@hookform/resolvers": "2.8.9",
"zod": "3.16.0"
```

Let's imagine we have this simple Zod schema:
```const schema = z
    .object({
        city: z.string().nullable(),
        street: z.string().nullable(),
        fullAddress: z.string().nullable(),
      })
    .transform((value) => ({
        ...value,
        fullAddress: `${value.city} ${value.street}`
    }));
```
and we use it with `react-hook-form`:
```
useForm({
        defaultValues: values,
        resolver: zodResolver(schema),
    });
```

Now when I'm changing city/street inputs I'm facing that despite `transform` returns correct values where `fullAddress` is equal to two concatenated field, input for `fullAddress` doesn't update its value.

The question is: how to update inputs for fields, whose value is tied to other fields values?
[CodeSandbox Example](https://codesandbox.io/s/rhf-zod-relations-m1om85) (please change fields and see console output)

**Comment by iambryanhaney** [comment, marked as answer]
Validation transformations don't affect the fields themselves; the data flows in one direction:

Fields (internal form state) -> submission / onChange / onBlur (depending on validation mode) -> resolver validation -> onValid / onInvalid

Transformed values are available post-validation, in the onValid callback.
